plugins/pulp_ostree/plugins/lib.py

```diff
@@ -374,7 +412,20 @@ def __init__(self, repository):
         """
         self.repository = repository
 
+    @property
+    def impl(self):
+        return self.repository.impl
+
     @wrapped
-    def generate(self):
+    def open(self):
+        """
+        Open the associated repository.
+
+        :raises LibError:
+        """
         self.repository.open()
-        self.repository.impl.regenerate_summary(None)
+
+    @wrapped
+    def generate(self):
+        self.open()
+        self.impl.regenerate_summary(None)
```

Comment:
This takes a `GCancellable` argument which is not optional on the RHEL7 versions of the libraries.